automates/text_reading/src/main/scala/org/clulab/aske/automates/mentions/CrossSentenceEventMention.scala

```diff
@@ -0,0 +1,76 @@
+                                 val foundBy: String,
+                                 val attachments: Set[Attachment]
+                               ) extends Mention {
+
+  def this(
+            label: String,
+            trigger: TextBoundMention,
+            arguments: Map[String, Seq[Mention]],
+            paths: Map[String, Map[Mention, SynPath]],
+            sentence: Int,
+            sentences: Seq[Int],
+            document: Document,
+            keep: Boolean,
+            foundBy: String
+          ) = this(Seq(label), mkTokenInterval(trigger, arguments), trigger, arguments, paths, sentence, sentences, document, keep, foundBy, Set.empty)
+
+  override def text: String = {
+    val sentenceAndMentionsSeq = (arguments
+      .values
+      .flatten ++ Seq(trigger))
+      .groupBy(_.sentence)
+      .toSeq
+      .sortBy(_._1) // sort by the sentence
+    // Since this is CrossSentence, there are at least two different sentences.
+    val firstSentence = sentenceAndMentionsSeq.head._1
+    val lastSentence = sentenceAndMentionsSeq.last._1
+    val perSentenceWords = sentenceAndMentionsSeq.map { case (sentence, mentions) =>
+      val sentenceWordArr = mentions.head.sentenceObj.raw //sentence as an array of tokens
+      //compile text of the CrossSentenceEventMention from parts of the sentences that the CrossSentenceEventMention spans
+      val words = sentence match { //sentence index (ordered)
+        case sentence if sentence == firstSentence =>
+          // in the first sentence the CrossSentenceEventMention spans, the part to return is the span from the start of the first argument or trigger of the event to the end of the sentence
+          // Although it doesn't matter much with a small collection, sorting one in its entirety just to extract
+          // an extreme value is inefficient.  So, a simple minBy is used.  Is there no minByBy?
+          val start = mentions.minBy(_.start).start
+          sentenceWordArr.drop(start)
+        case sentence if sentence == lastSentence =>
+          // in the last sentence the CrossSentenceEventMention spans, the part to return is the span from the beginning of the sentence to the end of the last argument or the trigger, whichever comes latest
+          // Although it may not be a problem in this context, the maximum end does not necessarily come from the
+          // mention with the maximum start.  Sometimes mentions overlap and they might conceivably be nested.
+          val end = mentions.maxBy(_.end).end
+          sentenceWordArr.take(end)
+        case _ =>
+          // if it's a middle sentence, the part to return is the whole sentence
+          sentenceWordArr
+      }
+
+      words
+    }
+    val text = perSentenceWords.flatten.mkString(" ")
+
+    text
+  }
+}
+
+object CrossSentenceEventMention // note: this seems like a necessary part - is there any way that this is incorporated into other part and deleted here?
```

Comment:
About the note: both you and I need to read a bit on classes vs objects in Scala to answer this :) Either case, for now, keep the object and delete the note